cmd/deploy/deploy.go

```diff
@@ -0,0 +1,173 @@
+	getSecrets  func(ctx context.Context) ([]string, error)
+	getManifest func(cwd, name, filename string) (*Manifest, error)
+
+	proxy      proxyInterface
+	kubeconfig kubeConfigHandler
+	executor   manifestExecutor
+}
+
+//Deploy deploys the okteto manifest
+func Deploy(ctx context.Context) *cobra.Command {
+	var filename string
+	var variables []string
+	var name string
+
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Executes the list of commands specified in the Okteto manifest",
+		Args:  utils.NoArgsAccepted("https://okteto.com/docs/reference/cli/#version"),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := contextCMD.Init(ctx); err != nil {
+				return err
+			}
+
+			cwd, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("failed to get the current working directory: %w", err)
+			}
+
+			if name == "" {
+				name = getName(cwd)
+			}
+
+			// Look for a free local port to start the proxy
+			port, err := model.GetAvailablePort("localhost")
+			if err != nil {
+				log.Errorf("could not find a free port to start proxy server: %s", err)
+				return err
+			}
+			log.Debugf("found available port %d", port)
+
+			// TODO for now, using self-signed certificates
+			cert, err := tls.X509KeyPair(cert, key)
+			if err != nil {
+				log.Errorf("could not read certificate: %s", err)
+				return err
+			}
+
+			// Generate a token for the requests done to the proxy
+			sessionToken := uuid.NewString()
+
+			c := &deployCommand{
+				getSecrets:  getOktetoSecretsAsEnvironmenVariables,
+				getManifest: getManifest,
+
+				kubeconfig: newKubeConfig(),
+				executor:   newExecutor(),
+				proxy: newProxy(proxyConfig{
+					port:         port,
+					certificates: []tls.Certificate{cert},
+					token:        sessionToken,
+				}),
+			}
+			return c.runDeploy(ctx, cwd, name, filename, variables)
+		},
+	}
+
+	cmd.Flags().StringVarP(&name, "name", "a", "", "application name")
+	cmd.Flags().StringVarP(&filename, "filename", "f", "", "path to the manifest file")
+	cmd.Flags().StringArrayVarP(&variables, "var", "v", []string{}, "set a pipeline variable (can be set more than once)")
+
+	return cmd
+}
+
+func (dc *deployCommand) runDeploy(ctx context.Context, cwd, name, filename string, variables []string) error {
+	clusterConfig, err := dc.kubeconfig.Read()
+	if err != nil {
+		log.Errorf("could not build kube config: %s", err)
+		return err
+	}
+
+	if err := dc.kubeconfig.Modify(ctx, dc.proxy.GetPort(), dc.proxy.GetToken()); err != nil {
+		log.Errorf("could not create temporal kubeconfig %s", err)
+		return err
+	}
+
+	// Read manifest file with the commands to be executed
+	m, err := dc.getManifest(cwd, name, filename)
+	if err != nil {
+		log.Errorf("could not find manifest file to be executed: %s", err)
+		return err
+	}
+
+	// Get secrets from API
+	sList, err := dc.getSecrets(ctx)
+	if err != nil {
+		log.Errorf("could not load Okteto Secrets: %s", err.Error())
+		sList = []string{}
+	}
```

Comment:
For the pipeline-installer-image I think this wouldn't be necessary since the installer would inject them when execute the command, but I guess that for the rest of cases we want to include secrets on the execution as we do for other commands

Reply:
It is already done when initializing the context, you can remove it